Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM python:3.10.4-slim-buster as base
+RUN pip install poetry
+EXPOSE 5000
+WORKDIR /app
+COPY . /app/
+RUN poetry install --no-root --no-dev

```

Comment:
Note that you are copying all files into the image, and then installing dependencies. If any file at all changes, then both these steps need to run again.

Alternatively, you could have three steps: copy in the poetry/pyproject files, install dependencies and then copy in everything else. If app files change (not poetry files), the second layer with installed dependencies is unaffected. That can be reused for the new image with just the final step running again - copying in app files.

This means that when dependencies don't change, rebuilds are faster and take up less space